language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/ResourcePathCompletionUtil.java

```diff
@@ -211,19 +212,47 @@ private static ResourceAccessPathPart getResourceAccessPartForSegment(PathSegmen
         String paramType = FunctionCompletionItemBuilder
                 .getFunctionParameterSyntax(pathParameterSymbol, context).orElse("");
         String computedSignature = "[" + paramType + "]";
-        String computedInsertText = "[${" + placeHolderIndex + ":" + defaultValue.orElse("") + "}]";
+        String computedInsertText =
+                typeSymbol.typeKind().equals(TypeDescKind.SINGLETON) ? "[" + defaultValue.orElse("") + "]" :
+                        "[${" + placeHolderIndex + ":" + defaultValue.orElse("") + "}]";
         ResourceAccessPathPart resourceAccessPathPart =
                 new ResourceAccessPathPart(computedInsertText, computedSignature);
         if (context.currentSemanticModel().isPresent() &&
-                context.currentSemanticModel().get().types().STRING.subtypeOf(typeSymbol)) {
-            resourceAccessPathPart.namedPathSignature = "<path>";
-            resourceAccessPathPart.namedPathInsertText = "${" + placeHolderIndex + ":" + "path" + "}";
-            resourceAccessPathPart.computedPathInsertText = "[${" + placeHolderIndex + ":" + "\"path\"" + "}]";
-            resourceAccessPathPart.isStringPathPram = true;
+                checkSubtype(typeSymbol, context.currentSemanticModel().get().types().STRING)) {
+            if (typeSymbol.typeKind().equals(TypeDescKind.SINGLETON)) {
+                resourceAccessPathPart.namedPathSignature =
+                        typeSymbol.signature().substring(1, typeSymbol.signature().length() - 1);
+                resourceAccessPathPart.namedPathInsertText = resourceAccessPathPart.namedPathSignature;
+            } else {
+                resourceAccessPathPart.namedPathSignature = "<path>";
+                resourceAccessPathPart.namedPathInsertText = "${" + placeHolderIndex + ":" + "path" + "}";
+                resourceAccessPathPart.computedPathInsertText = "[${" + placeHolderIndex + ":" + "\"path\"" + "}]";
+            }
+            resourceAccessPathPart.isStringPathParam = true;
         }
         return resourceAccessPathPart;
     }
 
+    private static boolean checkSubtype(TypeSymbol paramType, TypeSymbol stringType) {
```

Comment:
We can move this to CommonUtil class.

Reply:
Currently, the method is only used within the class.